tests/mpmc_channel.rs

```diff
@@ -918,6 +941,41 @@ macro_rules! gen_mpmc_tests {
                 assert_send_done(cx, &mut send_fut2, Ok(()));
                 assert_send_done(cx, &mut send_fut3, Ok(()));
             }
+
+            #[test]
+            fn stream_smoke_test() {
+                let channel = ChannelType::new();
+                let (waker, count) = new_count_waker();
+                let cx = &mut Context::from_waker(&waker);
+
+                let stream = channel.stream();
+                pin_mut!(stream);
+
+                assert!(stream.as_mut().poll_next(cx).is_pending());
+                assert_eq!(count, 0);
+
+                assert_send(cx, &channel, 1);
+
+                assert_next_done(cx, &mut stream, Some(1));
```

Comment:
You could try reading multiple values here. I guess just a loop which covers line 954-960 for some iterations might be fine.